app/models/invoice/data_presenter.rb

```diff
@@ -91,6 +91,14 @@ def display_shipment_amount_with_taxes
       Spree::Money.new(shipment.amount + shipment.additional_tax_total, currency:)
     end
 
+    def display_line_item_tax_rate(item)
+      all_tax_adjustments.select { |a|
+        a.adjustable.type == 'Spree::LineItem' && a.adjustable.id == item.id
+      }.map(&:originator).map { |tr|
```

Comment:
Couldn't you use a query here ? something like :
```ruby
all_tax_adjustments..where(adjustable_type: "Spree::LineItem").where(adjustable_id: item.id)
  .map(&:originator) 
  #  ...
```

Reply:
I don't think that's possible.
This code runs inside the data presenter. `all_tax_adjustments` represents an array inside the serialized invoice.

Reply:
I see, I always forget the data presenter is working on serialised data.